### DISCUSSION TITLE
Fzf-lua grep ... can be made case insensitive but not the preview window (the hl of the matched word)

### DISCUSSION THREAD

**Discussion by 1ngemar** [ibhagwan/fzf-lua#1563] [Q&A] [answered]
Hi all,

When running Fzf-lua grep ... I'm trying to have the builtin preview window also highlight the match found by Fzf-lua grep (rg) "smartmatch" setting.

Thus, if you Fzf-lua grep for "hello" and you have both "Hello" and "hello" in a file, only the "hello" match gets highlighted, even though the search engine (rg) finds them both and presents two rows in the fzf window to select.

![1rg](https://github.com/user-attachments/assets/76327146-187c-48a5-8681-eeaabd0e295a)
![2rg](https://github.com/user-attachments/assets/f2236548-3414-4bbc-9317-f2b67aba6764)

Any way forward in this?

**Comment by ibhagwan** [comment, marked as answer]
The highlight is performed via a `vim.regex:match_line` I'm not sure if there's a magic character equiavlent to rg's smart case, if there is I'd be happy to add it.

Otherwise it would mean i would have to modify the regex to `h -> [Hh]`  depending if the `--smart-case` flag is found, too prone to errors to be worth the effort IMHO.

**Comment by 1ngemar** [comment]
Ok, thanks.

Apparently, this is not trivial stuff; at least a bit over my head :) Let's put a pin in it for now. Not a big deal.

Thanks for a great plugin.
